### DISCUSSION TITLE
Is There a Way to Detect Auto-Recovery from Lost Steps, AKA Crash Detection?

### DISCUSSION THREAD

**Discussion by tperillo** [greghesp/ha-bambulab#572] [General]
I successfully set up my Bambu printer with Home Assistant, but I'm exploring how to enhance its functionality by pausing the printer if it detects a crash. The addition of a new smooth PEI plate and a bit of hairspray has significantly improved the adherence of small parts to the bed. Despite this, crashes can still occur, shifting the bed and causing layer shifts. The printer does recalibrate itself and resumes printing after a crash, but I'd prefer to manually check things over. My goal is to configure it to pause upon detecting a crash, giving me the opportunity to assess and either continue or cancel the print as needed.

In my initial attempts to address this, I configured home assistant to trigger alerts 'Hms errors started detecting problem' and 'Print error detected.' Unfortunately I don't think these are detecting the actual crashes. 

Thanks!

**Comment by AdrianGarside** [comment]
AFAIK there's no event that triggered for this scenario exposed in the mqtt payload.
